src/components/TextEditor.jsx

```
@@ -297,17 +431,21 @@ export default function TextEditor() {
+      // 入力操作のスナップショットをデバウンス（1秒間隔）
+      clearTimeout(snapshotTimer);
+      snapshotTimer = setTimeout(() => snapshot(), 1000);
     };
     const editor = editorRef.current;
     if (editor) {
       editor.addEventListener('input', handleInput);
-      return () => editor.removeEventListener('input', handleInput);
+      return () => { editor.removeEventListener('input', handleInput); clearTimeout(snapshotTimer); };
     }
-  }, []);
+  }, [snapshot]);
```

Biome (2.4.4)

[error] This variable is used before its declaration.

(lint/correctness/noInvalidUseBeforeDeclaration)